std/range/package.d

```diff
@@ -4756,21 +4755,28 @@ body
                                             cast(Value)(pastLast - (length - upper) * step),
                                             step);
         }
-        @property IndexType length() const
+        @property size_t length() const
         {
             if (step > 0)
             {
-                return unsigned((pastLast - current) / step);
+                return cast(size_t)unsigned((pastLast - current) / step);
             }
             else
             {
-                return unsigned((current - pastLast) / -step);
+                return cast(size_t)unsigned((current - pastLast) / -step);
             }
         }
 
         alias opDollar = length;
     }
 
+    if (step > 0)
+        assert(unsigned((end - begin) / step) <= size_t.max);
+    else if (step < 0)
+        assert(unsigned((end - begin) / -step) <= size_t.max);
+    //No else branch because if step = 0 it's either an error
+    //or begin == end and the range is empty
```

Comment:
Why not just move all of these into the precondition with the other assertion? Then you can even rewrite it in a way that the comment isn't necessary - like turn the assertion that's currently in the precondition into an else branch.